utils/build.ps1

```diff
@@ -1766,6 +1766,12 @@ function Build-FoundationMacros() {
     $InstallDir = "$($Arch.ToolchainInstallRoot)\usr"
   }
 
+  $SwiftSyntaxCMakeModules = if ($Build -and $HostArch -ne $BuildArch) {
```

Comment:
Can we not get away with `if ($Build)`? We will have the host build for distribution anyway.

Reply:
No, because `SwiftSyntax` is built part of `Build-Compilers` and in a normal (not cross-compiling) build, we only build in the host-arch binary cache (`Get-HostProjectCMakeModules Compilers`) and the build-arch binary cache `Get-BuildProjectCMakeModules Compilers` would be empty. If we just have `if ($Build)`, we would look for the SwiftSyntax build artifacts in the empty location. It's true that we build `FoundationMacros` twice and both for the build-arch and host-arch binary caches. But this is about the `SwiftSyntax`.

Reply:
(The current cross-compile CI build is fixed by line 1755 above)